solidity/contracts/bridge/Bridge.sol

```diff
@@ -795,10 +935,500 @@ contract Bridge is Ownable {
         return (inputTxHash, inputTxIndex, inputLength);
     }
 
-    // TODO It is possible a malicious wallet can sweep deposits that can not
-    //      be later proved on Ethereum. For example, a deposit with
-    //      an incorrect amount revealed. We need to provide a function for honest
-    //      depositors, next to sweep, to prove their swept balances on Ethereum
-    //      selectively, based on deposits they have earlier received.
-    //      (UPDATE PR #90: Is it still the case since amounts are inferred?)
+    /// @notice Requests redemption of the given amount from the specified
+    ///         wallet to the redeemer Bitcoin output script hash.
+    /// @param walletPubKeyHash The 20-byte wallet public key hash (computed
+    ///        using HASH160 opcode)
+    /// @param mainUtxo Data of the wallet's main UTXO, as currently known on
+    ///        the Ethereum chain
+    /// @param redeemerOutputHash The 20-byte (P2PKH, P2WPKH, or P2SH) or
+    ///        32-byte (P2WSH) output hash that will be used to lock
+    ///        redeemed BTC
+    /// @param amount Requested amount in satoshi. This is also the TBTC amount
+    ///        that is taken from redeemer's balance in the Bank upon request.
+    ///        Once the request is handled, the actual amount of BTC locked
+    ///        on the redeemer output hash will be always lower than this value
+    ///        since the treasury and Bitcoin transaction fees must be incurred.
+    ///        The minimal amount satisfying the request can be computed as:
+    ///        `amount - redemptionTreasuryFee - redemptionTxMaxFee`.
+    ///        Fees values are taken at the moment of request creation.
+    /// @dev Requirements:
+    ///      - Wallet behind `walletPubKeyHash` cannot be marked as faulty
```

Comment:
If we decide to introduce wallet states, let's remember to change it to:

```suggestion
    ///      - Wallet behind `walletPubKeyHash` must be active
```

Reply:
See a1b472e2